azure-functions/src/commands/run.rs

```diff
@@ -0,0 +1,399 @@
+
+struct HttpService(IpAddr, u16);
+
+impl Service<()> for HttpService {
+    type Response = TcpStream;
+    type Error = ::std::io::Error;
+    type Future = ConnectFuture;
+
+    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        Ok(().into())
+    }
+
+    fn call(&mut self, _: ()) -> Self::Future {
+        TcpStream::connect(&SocketAddr::new(self.0, self.1))
+    }
+}
+
+pub struct Run<'a> {
+    host: &'a str,
+    port: u16,
+    worker_id: &'a str,
+    //max_message_length: Option<u32>,
+}
+
+impl<'a> Run<'a> {
+    pub fn create_subcommand<'b>() -> App<'a, 'b> {
+        SubCommand::with_name("run")
+            .about("Runs the Rust language worker.")
+            .arg(
+                Arg::with_name("host")
+                    .long("host")
+                    .value_name("HOST")
+                    .help("The hostname of the Azure Functions Host.")
+                    .required(true),
+            )
+            .arg(
+                Arg::with_name("port")
+                    .long("port")
+                    .value_name("PORT")
+                    .help("The port of the Azure Functions Host.")
+                    .required(true),
+            )
+            .arg(
+                Arg::with_name("worker_id")
+                    .long("workerId")
+                    .value_name("WORKER_ID")
+                    .help("The worker ID to use when registering with the Azure Functions Host.")
+                    .required(true),
+            )
+            .arg(
+                Arg::with_name("request_id")
+                    .long("requestId")
+                    .value_name("REQUEST_ID")
+                    .help("The request ID to use when communicating with the Azure Functions Host.")
+                    .hidden(true)
+                    .required(true),
+            )
+            .arg(
+                Arg::with_name("max_message_length")
+                    .long("grpcMaxMessageLength")
+                    .value_name("MAXIMUM")
+                    .help("The maximum message length to use for gRPC messages."),
+            )
+    }
+
+    pub fn execute(&self, mut registry: Registry<'static>) -> Result<(), String> {
+        ctrlc::set_handler(|| {}).expect("failed setting SIGINT handler");
+
+        let uri = format!("http://{0}:{1}", self.host, self.port);
+        let (sender, receiver) = unbounded::<StreamingMessage>();
+
+        // Start by sending a start stream message to the channel
+        // This will be sent to the host upon connection
+        sender
+            .unbounded_send(StreamingMessage {
+                content: Some(Content::StartStream(StartStream {
+                    worker_id: self.worker_id.to_owned(),
+                })),
+                ..Default::default()
+            })
+            .unwrap();
+
+        println!(
+            "Connecting to Azure Functions host at {}:{}.",
+            self.host, self.port
+        );
```

Comment:
Can this just be

``` rust
println!("Connecting to Azure Functions host at {}", uri );
```

If so, I might do that and I might omit the trailing period. Some terminal environments are bad at identifying links and will include the period - making the URL resolve incorrectly. It might also be better if people are copying-pasting the link.

Reply:
Good point.  This was from the old lib.rs implementation that didn't have a uri just sitting there.  I'll make that change.

Reply:
Glad I could help. I'll add this contribution to my Connect for this quarter.